worlds/bomb_rush_cyberfunk/Rules.py

```diff
@@ -0,0 +1,1027 @@
+            lambda state: camera(state, player))
+        set_rule(multiworld.get_location("Pyramid Island: Polo pile 4", player),
+            lambda state: camera(state, player))
+        set_rule(multiworld.get_location("Pyramid Island: Maze glass Polo", player),
+            lambda state: camera(state, player))
+        set_rule(multiworld.get_location("Pyramid Island: Maze classroom Polo", player),
+            lambda state: camera(state, player))
+        set_rule(multiworld.get_location("Pyramid Island: Maze vent Polo", player),
+            lambda state: camera(state, player))
+        set_rule(multiworld.get_location("Pyramid Island: Big maze Polo", player),
+            lambda state: camera(state, player))
+        set_rule(multiworld.get_location("Pyramid Island: Maze desk Polo", player),
+            lambda state: camera(state, player))
+        set_rule(multiworld.get_location("Pyramid Island: Maze forklift Polo", player),
+            lambda state: camera(state, player))
+
+    # mataan
+    set_rule(multiworld.get_location("Mataan: Race challenge reward", player),
+        lambda state: mataan_challenge1(state, player, limit, glitched))
+    set_rule(multiworld.get_location("Mataan: Wallrunning challenge reward", player),
+        lambda state: mataan_challenge2(state, player, limit, glitched))
+    set_rule(multiworld.get_location("Mataan: Score challenge reward", player),
+        lambda state: mataan_challenge3(state, player))
+    if photos:
+        set_rule(multiworld.get_location("Mataan: Trash Polo", player),
+            lambda state: camera(state, player))
+        set_rule(multiworld.get_location("Mataan: Shopping Polo", player),
+            lambda state: camera(state, player))
+
+    # events
+    set_rule(multiworld.get_location("Versum Hill: Complete Chapter 1", player),
+        lambda state: versum_hill_crew_battle(state, player, limit, glitched))
+    set_rule(multiworld.get_location("Brink Terminal: Complete Chapter 2", player),
+        lambda state: brink_terminal_crew_battle(state, player, limit, glitched))
+    set_rule(multiworld.get_location("Millennium Mall: Complete Chapter 3", player),
+        lambda state: millennium_mall_crew_battle(state, player, limit, glitched))
+    set_rule(multiworld.get_location("Pyramid Island: Complete Chapter 4", player),
+        lambda state: pyramid_island_crew_battle(state, player, limit, glitched))
+    set_rule(multiworld.get_location("Defeat Faux", player),
+        lambda state: mataan_faux(state, player, limit, glitched))
+    
+    if extra:
+        add_rule(multiworld.get_location("Defeat Faux", player),
+            lambda state: rep(state, player, 1000))
+
+    
+    # graffiti spots
+    spots: int = 0
+    while spots < 385:
+        spots += 5
+        set_rule(multiworld.get_location(f"Tagged {spots} Graffiti Spots", player),
+            lambda state: graffiti_spots(state, player, movestyle, limit, glitched, spots))
```

Comment:
i believe this is making all tagged spots locations out of logic until you can access 380 spots because of how lambdas and variables work in loops
changing to lambda state, spots=spots: should fix that

Reply:
Ah, good to know